### DISCUSSION TITLE
License/open source

### DISCUSSION THREAD

**Discussion by mika76** [twinbasic/twinbasic#135] [General]
From the documentation page:
> This is a commercial project, but we are not asking for money at this stage. Additionally, the 32-bit Windows version of the project will be available completely free of charge. Developers that need to build 64-bit binaries will need to purchase a licence. Further details on this will be provided soon.

I'm just interested in whether there will be any open source side to this project and will other contributors be allowed?

I find closed-source, single dev development tools do not really apply to the modern market. It seems to me to be far better to create some sort of commercial services, learning assets, and take corporate sponsorships with a lively community of users and contributors.

**Comment by WaynePhillipsEA** [comment]
At the moment, the plan is that the main compiler will be closed source.   We will, however, be opening up several parts of the compiler through the use of extensions.  For example, there is a proposed analysis extension API #51 where we will be opening up the compiler AST etc for extensions to consume.
